client/src/components/Bills/BillForm.js

```diff
@@ -0,0 +1,194 @@
+import { useState } from "react";
+import ContactDropDown from "../DropDowns/ContactDropList";
+
+const emptyBill = {
+  transaction_date: "",
+  subtotal: null,
+  tax_rate: 0,
+  tax_amount: null,
+  tip_rate: null,
+  tip_total: 0,
+  who_paid: null,
+  paid_up: null,
+  notes: "",
+  full_total: 0,
+};
+
+const BillForm = (props) => {
+  //An initial student if there is one in props
+  const { initialBill } = props;
+
+  // Initial State
+  const [bill, setBill] = useState(initialBill || emptyBill);
+
+  //create functions that handle the event of the user typing into the form
+  const handleChange = (event) => {
+    const name = event.target.name;
+    const value = event.target.value;
+    setBill((bill) => ({ ...bill, [name]: value }));
+    console.log("Bill, client side: " + name + ", value: " + value);
+  };
+
+  //A function to handle the POST request
+  const postNewBill = (newBill) => {
+    return fetch("http://localhost:5005/db/bills", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(newBill),
+    })
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        console.log("From the bill add ", data);
+        // props.addPost(data);
+      });
+  };
+
+  //A function to handle the PUT request
+  const updateBillInfo = async (existingBill) => {
+    return fetch(`http://localhost:5005/db/bills/${existingBill.bill_id}`, {
+      method: "PUT",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(existingBill),
+    })
+      .then((response) => {
+        return response.json();
+      })
+      .then((data) => {
+        console.log("The updated bill info: ", data);
+        props.updateContact(data);
+      });
```

Comment:
❗ this should (in real code) have a `catch()` clause to capture errors.

Reply:
Did this for the POSTs, PUTs, and handleSubmits.